azurerm/resource_arm_database_vulnerability_assessment_rule_baseline.go

```diff
@@ -0,0 +1,238 @@
+)
+
+func resourceArmDatabaseVulnerabilityAssessmentRuleBaseline() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDatabaseVulnerabilityAssessmentRuleBaselineCreateUpdate,
+		Read:   resourceArmDatabaseVulnerabilityAssessmentRuleBaselineRead,
+		Update: resourceArmDatabaseVulnerabilityAssessmentRuleBaselineCreateUpdate,
+		Delete: resourceArmDatabaseVulnerabilityAssessmentRuleBaselineDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"server_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateMsSqlServerName,
+			},
+
+			"database_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateMsSqlDatabaseName,
+			},
+
+			"rule_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"baseline_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(sql.VulnerabilityAssessmentPolicyBaselineNameDefault),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.VulnerabilityAssessmentPolicyBaselineNameDefault),
+					string(sql.VulnerabilityAssessmentPolicyBaselineNameMaster),
+				}, true),
+			},
+
+			"baseline_result": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"result": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmDatabaseVulnerabilityAssessmentRuleBaselineCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).databaseVulnerabilityAssessmentRuleBaselinesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Vulnerability Assessment Rule Baselines creation.")
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	serverName := d.Get("server_name").(string)
+	databaseName := d.Get("database_name").(string)
+	ruleId := d.Get("rule_id").(string)
+	baselineName := expandBaselineName(d.Get("baseline_name").(string))
+	parameters := expandBaselineResults(d.Get("baseline_result").(*schema.Set))
+
+	result, err := client.CreateOrUpdate(ctx, resourceGroupName, serverName, databaseName, ruleId, *baselineName, *parameters)
+
+	if err != nil {
+		return fmt.Errorf("error updataing database vulnerability assessment rule baseline: %s", err)
+	}
+
+	id, err := parseAzureResourceID(*result.ID)
+
+	if err != nil {
+		return err
+	}
+
+	// Currently REST API always returns "Default" as baseline that's why the Id is created manually here as a workaround
```

Comment:
Is there an open issue on the SDK to track this? if not might be worth opening one

Reply:
This is not an issue of the SDK, it's an issue of the Azure REST api. I contacted the Azure support about this. They confirmed it's a bug and will be fixed in the future. Should we still open an issue on the SDK to track this?

Reply:
🤔 

We usually just throw issues there so they can route it to the correct team (rather then us trying to track down the right contact)

But this does post a little bit of an issue if we are going to support multiple database types. I think you'll need to change the path depending on if its mysql/mssql/postgres ect?

Reply:
I created an Issue https://github.com/Azure/azure-sdk-for-go/issues/5435

Sorry I was wrong, MS SQL is the only supported database type.